### PR TITLE
Adding BioRxiv and MedRxiv.

### PR DESCRIPTION
**Pull request by AikyaKumar**
# For authors

## Description
This PR introduces a unified tool for retrieving metadata and PDF URLs for preprints from bioRxiv and medRxiv using their DOIs. The tool performs the following key functions:

- DOI normalization (e.g., stripping version suffixes like v1, v2) to ensure compatibility with the API.
- Fetches article metadata (title, authors, abstract, date) from the [api.biorxiv.org](https://api.biorxiv.org/) endpoint, which supports both bioRxiv and medRxiv preprints.
- Constructs a direct PDF URL based on the clean DOI.
- Determines source ("biorxiv" or "medrxiv") dynamically based on the DOI prefix.

## Fixes 
No fixed done. This is a new addition in issue #173.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Unit tests written using unittest and unittest.mock for both bioRxiv and medRxiv DOIs.
- Tests cover successful metadata retrieval, missing metadata scenarios, and PDF fallback URL behavior.
- Manual testing with DOIs from both sources to verify correct source inference and PDF generation.

Example test cases covered:

- bioRxiv paper with full metadata and link
- medRxiv paper without a direct PDF link (fallback construction)
- Invalid or unknown DOI returns graceful error

## Checklist

- My code follows the style guidelines mentioned in the Code/DevOps guides
-  I have commented my code, particularly in hard-to-understand areas
-  I have made corresponding changes to the documentation (e.g. MkDocs)
-  My changes generate no new warnings
-  I have added or updated tests (in the tests folder) that prove my fix is effective or that my feature works
-  New and existing tests pass locally with my changes
-  Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

File: aiagents4pharma/talk2scholars/tests/test_paper_download_biorxiv.py
Comment (ansh-info) [inline aiagents4pharma/talk2scholars/tests/test_paper_download_biorxiv.py:1]: Good work with the tests 👍

File: aiagents4pharma/talk2scholars/tests/test_paper_download_medrxiv.py
Comment (ansh-info) [inline aiagents4pharma/talk2scholars/tests/test_paper_download_medrxiv.py:1]: Same here 👍

File: aiagents4pharma/talk2scholars/tools/paper_download/download_biorxiv_input.py
Comment (ansh-info) [inline aiagents4pharma/talk2scholars/tools/paper_download/download_biorxiv_input.py:1]: Good work, Nice Implementation ✅

File: aiagents4pharma/talk2scholars/tools/s2/utils/single_helper.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ def _filter_papers(self) -> None:
                 ],
                 "URL": paper.get("url", "N/A"),
                 "arxiv_id": paper.get("externalIds", {}).get("ArXiv", "N/A"),
+                "doi": paper.get("externalIds", {}).get("DOI", "N/A"),
```
Comment (ansh-info) [inline aiagents4pharma/talk2scholars/tools/s2/utils/single_helper.py:139]: Same here

**Review by ansh-info — APPROVED** [review]
Great Job on implementing both the tools in such short time 👍
I just have few comments related to the param you fetch in s2's helpers, and please consider dropping those column/param in the `streamlit_utils`

**Comment by ansh-info** [comment]
Please also take care of the tests and code coverage 👍

File: aiagents4pharma/talk2scholars/tools/s2/utils/single_helper.py (hunk 1 above)
Comment (AikyaKumar) [reply aiagents4pharma/talk2scholars/tools/s2/utils/single_helper.py:139]: I made corrections on this.
